plugins/nimble/nimble.go

```diff
@@ -0,0 +1,27 @@
+package nimble
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func nimCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "nimble",
+		Runs:    []string{"nimble"},
+		DocsURL: sdk.URL("https://github.com/nim-lang/nimble"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.IfAny(
+				needsauth.ForCommand("publish"),
+			),
+			needsauth.NotForHelp(),
+		),
+		Uses: []schema.CredentialUsage{
+			{
+				Name: credname.PersonalAccessToken,
```

Comment:
If we go on the route mentioned in the other comment, here we could just specify `Plugin: gh`, `Name: credname.PersonalAccessToken` and `Provisioner: the provisioner you created`.

Reply:
I'm all for using the standardised plugin!  The [schema suggests](https://pkg.go.dev/github.com/1Password/shell-plugins/sdk/schema#CredentialUsage) this needs to be a string for the plugin, e.g. `Plugin: "github"`?
I can't seem to see other plugins using it for an example, so any help would be massively appreciated.

As I've not added a custom provisioner, would something along the lines of this be close?
```go
...

Uses: []schema.CredentialUsage{
    {
      Plugin: "github",
      Name:   credname.PersonalAccessToken,
    },
},
```

Thanks in advance

Reply:
Yes, this is right!

The only extra thing is that you need to define a custom provisioner for this credential usage, as you noted in this other thread's [reply](https://github.com/1Password/shell-plugins/pull/255#discussion_r1183053495). 

So this should work:

```
Uses: []schema.CredentialUsage{
    {
      Plugin: "github",
      Name:   credname.PersonalAccessToken,
      Provisioner: provision.EnvVars(nimbleDefaultEnvVarMapping),
    },
},

...

var nimbleDefaultEnvVarMapping = map[string]sdk.FieldName{
	"NIMBLE_GITHUB_API_TOKEN": fieldname.Token,
}
```

while everything else in `nimble/personal_access_token.go` can be safely removed.

You could then remove the provisioner test case entirely or modify it to test this custom provisioner `provision.EnvVars(nimbleDefaultEnvVarMapping)`

Reply:
Let's add the provisioner here in the credential usage, as we discussed before